specification/authorization/resource-manager/Microsoft.Authorization/preview/2021-01-01-preview/authorization-RoleAssignmentApprovalCalls.json

```diff
@@ -0,0 +1,844 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2021-01-01-preview",
+    "title": "AuthorizationManagementClient",
+    "description": "Request Approvals service provides the workflow for running request approvals on different kind of resources."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.Authorization/operations": {
+      "get": {
+        "operationId": "Operations_List",
+        "x-ms-examples": {},
+        "description": "Lists the operations available from this provider.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Describe the result of a successful operation.",
+            "schema": {
+              "$ref": "#/definitions/OperationListResult"
+            }
+          },
+          "default": {
+            "description": "Unexpected error",
+            "schema": {
+              "$ref": "#/definitions/ErrorDefinition"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Authorization/roleAssignmentApprovals": {
+      "get": {
+        "tags": [
+          "ListRoleAssignmentApprovals"
+        ],
+        "operationId": "RoleAssignmentApproval_List",
+        "x-ms-examples": {
+          "GetAccessReviews": {
+            "$ref": "./examples/GetRoleAssignmentApprovals.json"
+          }
+        },
+        "description": "Get role assignment approvals.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Describe the result of a successful operation.",
+            "schema": {
+              "$ref": "#/definitions/RoleAssignmentApprovalListResult"
+            }
+          },
+          "default": {
+            "description": "Unexpected error",
+            "schema": {
+              "$ref": "#/definitions/ErrorDefinition"
```

Comment:
>      "$ref": "#/definitions/ErrorDefinition" [](start = 8, length = 45)

Consider using common error model defined in types.json: [https://github.com/Azure/azure-rest-api-specs/blob/44ec11c74634f1afa5bbe0d3f2678b7f819ebf8e/specification/common-types/resource-management/v2/types.json](https://github.com/Azure/azure-rest-api-specs/blob/44ec11c74634f1afa5bbe0d3f2678b7f819ebf8e/specification/common-types/resource-management/v2/types.json). One example is : [https://github.com/Azure/azure-rest-api-specs/blob/18942e7ffaf9bc39da616397b4a6b8a191a00ea1/specification/maps/resource-manager/Microsoft.Maps/preview/2020-02-01-preview/maps-management.json#L88](https://github.com/Azure/azure-rest-api-specs/blob/18942e7ffaf9bc39da616397b4a6b8a191a00ea1/specification/maps/resource-manager/Microsoft.Maps/preview/2020-02-01-preview/maps-management.json#L88)

Reply:
I copied this format from other APIs in Microsoft.Authorization RP. Do you think it's better to align with other APIs in the same RP?

Reply:
Filiz is right that the shared definition is the preferred approach, but for now let's align with the current APIs.

---
In reply to: [551663145](https://github.com/Azure/azure-rest-api-specs/pull/12108#discussion_r551663145) [](ancestors = 551663145)